### ISSUE TITLE
Replicant pod does not spin up when using Helm to deploy EMQX resource

### ISSUE THREAD

**Issue by Jaxwood** [emqx/emqx-operator#920]
**Describe the bug**
Replicant pod does not spin up when using Helm to deploy EMQX resource. Is stuck in `CoreNodesProgressing` state.

**To Reproduce**
- Create Helm chart with only EMQx resource inside, e.g.
```yaml
apiVersion: apps.emqx.io/v2beta1
kind: EMQX
metadata:
  name: emqx
  namespace: {{ .Release.Namespace }}
spec:
  image: {{ include "emqx.image" . }}
  coreTemplate:
    spec:
      replicas: 3
    replicantTemplate:
      spec:
        replicas: 3
```
- Deploy using helm, e.g. `helm upgrade --install -f values.yaml -n foo emqx .`

**Expected behavior**
EMQx core and replicant pods spin up.

**Anything else we need to know?**:
Using 5.1.4 version of EMQx image
Plain manifest files works fine.

**Environment details:**:
- Kubernetes version:
1.25.6

- Cloud-provider/provisioner:
AKS

- emqx-operator version:
2.2.1-rc.3

- Install method: e.g. helm/static manifests
Helm

**EMQX Status**
```text
NAME   STATUS                 AGE
emqx   CoreNodesProgressing   2m5s
```
**Core Logs**
```text
EMQX_API_KEY__BOOTSTRAP_FILE [api_key.bootstrap_file]: /opt/emqx/data/bootstrap_api_key
EMQX_RPC__PORT_DISCOVERY [rpc.port_discovery]: manual
EMQX_CLUSTER__DNS__RECORD_TYPE [cluster.dns.record_type]: srv
EMQX_CLUSTER__DNS__NAME [cluster.dns.name]: emqx-headless.ccoe.svc.cluster.local
EMQX_CLUSTER__DISCOVERY_STRATEGY [cluster.discovery_strategy]: dns
EMQX_NODE__ROLE [node.role]: core
EMQX_NODE__DATA_DIR [node.data_dir]: data
EMQX_NODE__COOKIE [node.cookie]: ******
EMQX_NODE__NAME [node.name]: emqx@emqx-core-7d8b454dd5-0.emqx-headless.ccoe.svc.cluster.local
Listener ssl:default on 0.0.0.0:8883 started.
Listener tcp:default on 0.0.0.0:1883 started.
Listener ws:default on 0.0.0.0:8083 started.
Listener wss:default on 0.0.0.0:8084 started.
Listener http:dashboard on :18083 started.
EMQX 5.1.4 is running now!
```

**Operator logs (with debug enabled)**
```text
I0813 09:45:25.769473       1 request.go:601] Waited for 1.043159156s due to client-side throttling, not priority and fairness, request: GET:https://10.231.87.1:443/apis/notification.toolkit.fluxcd.io/v1beta2?timeout=32s
2023-08-13T09:45:26Z	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": ":8080"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxPlugin", "path": "/mutate-apps-emqx-io-v1beta4-emqxplugin"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/mutate-apps-emqx-io-v1beta4-emqxplugin"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxPlugin", "path": "/validate-apps-emqx-io-v1beta4-emqxplugin"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate-apps-emqx-io-v1beta4-emqxplugin"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/convert"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Conversion webhook enabled	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxPlugin"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxBroker", "path": "/mutate-apps-emqx-io-v1beta4-emqxbroker"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/mutate-apps-emqx-io-v1beta4-emqxbroker"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxBroker", "path": "/validate-apps-emqx-io-v1beta4-emqxbroker"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate-apps-emqx-io-v1beta4-emqxbroker"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Conversion webhook enabled	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxBroker"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxEnterprise", "path": "/mutate-apps-emqx-io-v1beta4-emqxenterprise"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/mutate-apps-emqx-io-v1beta4-emqxenterprise"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxEnterprise", "path": "/validate-apps-emqx-io-v1beta4-emqxenterprise"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate-apps-emqx-io-v1beta4-emqxenterprise"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Conversion webhook enabled	{"GVK": "apps.emqx.io/v1beta4, Kind=EmqxEnterprise"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a mutating webhook	{"GVK": "apps.emqx.io/v2beta1, Kind=EMQX", "path": "/mutate-apps-emqx-io-v2beta1-emqx"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/mutate-apps-emqx-io-v2beta1-emqx"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "apps.emqx.io/v2beta1, Kind=EMQX", "path": "/validate-apps-emqx-io-v2beta1-emqx"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate-apps-emqx-io-v2beta1-emqx"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Conversion webhook enabled	{"GVK": "apps.emqx.io/v2beta1, Kind=EMQX"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "apps.emqx.io/v2beta1, Kind=Rebalance"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "apps.emqx.io/v2beta1, Kind=Rebalance", "path": "/validate-apps-emqx-io-v2beta1-rebalance"}
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate-apps-emqx-io-v2beta1-rebalance"}
2023-08-13T09:45:26Z	INFO	controller-runtime.builder	Conversion webhook enabled	{"GVK": "apps.emqx.io/v2beta1, Kind=Rebalance"}
2023-08-13T09:45:26Z	INFO	setup	starting manager
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook.webhooks	Starting webhook server
2023-08-13T09:45:26Z	INFO	Starting server	{"kind": "health probe", "addr": "[::]:8081"}
2023-08-13T09:45:26Z	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
I0813 09:45:26.381268       1 leaderelection.go:248] attempting to acquire leader lease kube-addons/19fd6fcc.emqx.io...
2023-08-13T09:45:26Z	INFO	controller-runtime.certwatcher	Updated current TLS certificate
2023-08-13T09:45:26Z	INFO	controller-runtime.webhook	Serving webhook server	{"host": "", "port": 9443}
2023-08-13T09:45:26Z	INFO	controller-runtime.certwatcher	Starting certificate watcher
2023-08-13T09:45:56Z	DEBUG	controller-runtime.webhook.webhooks	received request	{"webhook": "/mutate-apps-emqx-io-v2beta1-emqx", "UID": "2b9e47d2-8e46-4f97-9edd-d1365d08bda8", "kind": "apps.emqx.io/v2beta1, Kind=EMQX", "resource": {"group":"apps.emqx.io","version":"v2beta1","resource":"emqxes"}}
2023-08-13T09:45:56Z	INFO	emqx-resource	default	{"name": "emqx"}
2023-08-13T09:45:56Z	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/mutate-apps-emqx-io-v2beta1-emqx", "code": 200, "reason": "", "UID": "2b9e47d2-8e46-4f97-9edd-d1365d08bda8", "allowed": true}
2023-08-13T09:45:56Z	DEBUG	controller-runtime.webhook.webhooks	received request	{"webhook": "/validate-apps-emqx-io-v2beta1-emqx", "UID": "de3847a4-edaf-41f9-b2bc-8796ce79247c", "kind": "apps.emqx.io/v2beta1, Kind=EMQX", "resource": {"group":"apps.emqx.io","version":"v2beta1","resource":"emqxes"}}
2023-08-13T09:45:56Z	INFO	emqx-resource	validate create	{"name": "emqx"}
2023-08-13T09:45:56Z	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/validate-apps-emqx-io-v2beta1-emqx", "code": 200, "reason": "", "UID": "de3847a4-edaf-41f9-b2bc-8796ce79247c", "allowed": true}
I0813 09:45:57.467564       1 leaderelection.go:258] successfully acquired lease kube-addons/19fd6fcc.emqx.io
2023-08-13T09:45:57Z	DEBUG	events	Normal	{"object": {"kind":"Lease","namespace":"kube-addons","name":"19fd6fcc.emqx.io","uid":"a084e5db-525b-4b7b-b397-cd512a9cc4ba","apiVersion":"coordination.k8s.io/v1","resourceVersion":"1329592134"}, "reason": "LeaderElection", "message": "emqx-operator-controller-manager-6c95689b8-xmpxp_31d8099e-4fde-485f-a3cd-d9b5c57162cd became leader"}
2023-08-13T09:45:57Z	INFO	Starting EventSource	{"controller": "emqxbroker", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxBroker", "source": "kind source: *v1beta4.EmqxBroker"}
2023-08-13T09:45:57Z	INFO	Starting Controller	{"controller": "emqxbroker", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxBroker"}
2023-08-13T09:45:57Z	INFO	Starting EventSource	{"controller": "emqxplugin", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxPlugin", "source": "kind source: *v1beta4.EmqxPlugin"}
2023-08-13T09:45:57Z	INFO	Starting Controller	{"controller": "emqxplugin", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxPlugin"}
2023-08-13T09:45:57Z	INFO	Starting EventSource	{"controller": "emqx", "controllerGroup": "apps.emqx.io", "controllerKind": "EMQX", "source": "kind source: *v2beta1.EMQX"}
2023-08-13T09:45:57Z	INFO	Starting Controller	{"controller": "emqx", "controllerGroup": "apps.emqx.io", "controllerKind": "EMQX"}
2023-08-13T09:45:57Z	INFO	Starting EventSource	{"controller": "emqxenterprise", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxEnterprise", "source": "kind source: *v1beta4.EmqxEnterprise"}
2023-08-13T09:45:57Z	INFO	Starting EventSource	{"controller": "rebalance", "controllerGroup": "apps.emqx.io", "controllerKind": "Rebalance", "source": "kind source: *v2beta1.Rebalance"}
2023-08-13T09:45:57Z	INFO	Starting Controller	{"controller": "emqxenterprise", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxEnterprise"}
2023-08-13T09:45:57Z	INFO	Starting Controller	{"controller": "rebalance", "controllerGroup": "apps.emqx.io", "controllerKind": "Rebalance"}
2023-08-13T09:45:57Z	INFO	Starting workers	{"controller": "emqxbroker", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxBroker", "worker count": 1}
2023-08-13T09:45:57Z	INFO	Starting workers	{"controller": "emqxenterprise", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxEnterprise", "worker count": 1}
2023-08-13T09:45:57Z	INFO	Starting workers	{"controller": "emqx", "controllerGroup": "apps.emqx.io", "controllerKind": "EMQX", "worker count": 1}
2023-08-13T09:45:57Z	INFO	Starting workers	{"controller": "rebalance", "controllerGroup": "apps.emqx.io", "controllerKind": "Rebalance", "worker count": 1}
2023-08-13T09:45:57Z	INFO	Starting workers	{"controller": "emqxplugin", "controllerGroup": "apps.emqx.io", "controllerKind": "EmqxPlugin", "worker count": 1}
2023-08-13T09:46:00Z	INFO	KubeAPIWarningLogger	unknown field "spec.coreTemplate.metadata.creationTimestamp"
2023-08-13T09:46:00Z	INFO	KubeAPIWarningLogger	unknown field "spec.dashboardServiceTemplate.metadata.creationTimestamp"
2023-08-13T09:46:00Z	INFO	KubeAPIWarningLogger	unknown field "spec.listenersServiceTemplate.metadata.creationTimestamp"
2023-08-13T09:46:00Z	INFO	KubeAPIWarningLogger	unknown field "spec.replicantTemplate.metadata.creationTimestamp"
2023-08-13T09:46:01Z	DEBUG	events	Warning	{"object": {"kind":"EMQX","namespace":"ccoe","name":"emqx","uid":"88c08c29-c81c-4c3c-a539-ee03ce975eda","apiVersion":"apps.emqx.io/v2beta1","resourceVersion":"1329592307"}, "reason": "FailedToGetNodeStatuses", "message": "failed to get node statues by API: failed to get API http://10.231.84.191:18083/api/v5/nodes: failed to request API: Get \"http://10.231.84.191:18083/api/v5/nodes\": dial tcp 10.231.84.191:18083: connect: connection refused"}
2023-08-13T09:46:03Z	DEBUG	events	Warning	{"object": {"kind":"EMQX","namespace":"ccoe","name":"emqx","uid":"88c08c29-c81c-4c3c-a539-ee03ce975eda","apiVersion":"apps.emqx.io/v2beta1","resourceVersion":"1329592388"}, "reason": "FailedToGetNodeStatuses", "message": "failed to get node statues by API: failed to get API http://10.231.84.117:18083/api/v5/nodes: failed to request API: Get \"http://10.231.84.117:18083/api/v5/nodes\": dial tcp 10.231.84.117:18083: connect: connection refused"}
2023-08-13T09:46:04Z	DEBUG	events	Warning	{"object": {"kind":"EMQX","namespace":"ccoe","name":"emqx","uid":"88c08c29-c81c-4c3c-a539-ee03ce975eda","apiVersion":"apps.emqx.io/v2beta1","resourceVersion":"1329592388"}, "reason": "FailedToGetNodeStatuses", "message": "failed to get node statues by API: failed to get API http://10.231.84.115:18083/api/v5/nodes: failed to request API: Get \"http://10.231.84.115:18083/api/v5/nodes\": dial tcp 10.231.84.115:18083: connect: connection refused"}
2023-08-13T09:46:05Z	DEBUG	events	Warning	{"object": {"kind":"EMQX","namespace":"ccoe","name":"emqx","uid":"88c08c29-c81c-4c3c-a539-ee03ce975eda","apiVersion":"apps.emqx.io/v2beta1","resourceVersion":"1329592388"}, "reason": "FailedToGetNodeStatuses", "message": "failed to get node statues by API: failed to get API http://10.231.84.115:18083/api/v5/nodes: failed to request API: Get \"http://10.231.84.115:18083/api/v5/nodes\": dial tcp 10.231.84.115:18083: connect: connection refused"}
```

**Comment by Rory-Z** [comment]
Hi, @Jaxwood Sorry for delay, the EMQX Operator 2.2.1 is released, could you please try 2.2.1 ?

**Comment by Jaxwood** [comment]
> Hi, @Jaxwood Sorry for delay, the EMQX Operator 2.2.1 is released, could you please try 2.2.1 ?

Working fine now. Thanks.
